pkg/networkservice/common/begin/event_factory_client_test.go

```diff
@@ -130,6 +132,40 @@ func TestRefreshDuringClose_Client(t *testing.T) {
 	eventFactoryCl.callRefresh()
 }
 
+// This test checks if the timeout for the Request/Close called from the event factory is correct
+func TestContextTimeout_Client(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Add clockMock to the context
+	clockMock := clockmock.New(ctx)
+	ctx = clock.WithClock(ctx, clockMock)
+
+	ctx, cancel = context.WithDeadline(ctx, clockMock.Now().Add(time.Second*3))
+	defer cancel()
+
+	eventFactoryCl := &eventFactoryClient{}
+	client := chain.NewNetworkServiceClient(
+		begin.NewClient(),
+		eventFactoryCl,
+		&delayedNSEClient{t: t, clock: clockMock},
+	)
+
+	// Do Request
+	request := testRequest("1")
+	conn, err := client.Request(ctx, request.Clone())
+	assert.NotNil(t, t, conn)
```

Comment:
Why don't we use `require` package here?

Reply:
Replaced with `require`